executor/show.go

```diff
@@ -606,6 +606,7 @@ func (e *ShowExec) fetchShowWarnings() error {
 	for _, warn := range warns {
 		datums := make([]types.Datum, 3)
 		datums[0] = types.NewStringDatum("Warning")
+		warn = errors.Cause(warn)
```

Comment:
warn maybe return by errors.Trace(err), without this line, warn is not a terror.Error type. @XuHuaiyu